nano/node/backlog_population.cpp

```diff
@@ -0,0 +1,90 @@
+#include <nano/node/backlog_population.hpp>
+#include <nano/node/node.hpp>
+
+nano::backlog_population::backlog_population (nano::node & node_a) :
+	node{ node_a }
+{
+}
+
+nano::backlog_population::~backlog_population ()
+{
+	stop ();
+	if (thread.joinable ())
+	{
+		thread.join ();
+	}
+}
+
+void nano::backlog_population::start ()
+{
+	if (!thread.joinable ())
+	{
+		thread = std::thread{ [this] () { run (); } };
+	}
+}
+
+void nano::backlog_population::stop ()
+{
+	nano::unique_lock<nano::mutex> lock{ mutex };
+	stopped = true;
+	notify ();
+}
+
+void nano::backlog_population::trigger ()
+{
+	nano::unique_lock<nano::mutex> lock{ mutex };
+	triggered = true;
+	notify ();
+}
+
+void nano::backlog_population::notify ()
+{
+	condition.notify_all ();
+}
+
+bool nano::backlog_population::predicate () const
+{
+	return triggered;
+}
+
+void nano::backlog_population::run ()
+{
+	nano::thread_role::set (nano::thread_role::name::backlog_population);
+	nano::unique_lock<nano::mutex> lock{ mutex };
+	while (!stopped)
+	{
+		if (predicate () || (node.config.frontiers_confirmation != nano::frontiers_confirmation_mode::disabled))
```

Comment:
Why is this check 
```node.config.frontiers_confirmation != nano::frontiers_confirmation_mode::disabled```
not inside `predicate()`? I think it should be either inside predicate or the check should be outside the loop. Currently, the loop checks for this at every loop pass but it won't consider it for wakeup from sleep, which seems inconsistent.